main.go

```diff
@@ -1,32 +1,109 @@
 package main
 
 import (
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/urfave/cli/v2"
 	"go.uber.org/zap"
 
 	"github.com/Zamerykanizowana/replicated-file-system/config"
 	"github.com/Zamerykanizowana/replicated-file-system/logging"
+	"github.com/Zamerykanizowana/replicated-file-system/p2p"
 	"github.com/Zamerykanizowana/replicated-file-system/rfs"
 )
 
-var appConfig config.Config
-
 var (
 	branch = "unknown-branch"
 	commit = "unknown-commit"
 )
 
 func main() {
 	logging.Configure()
-	appConfig = config.ReadConfig()
+	cfg := mustReadConfig()
+
+	app := &cli.App{
+		Name:     "rfs",
+		HelpName: "Replicated file system using FUSE bindings and peer-2-peer architecture",
+		Flags:    flags(),
+		Commands: cli.Commands{
+			{
+				Name:   "p2p",
+				Action: func(context *cli.Context) error { return runP2P(cfg) },
```

Comment:
Question: is `context` passed here in order for cancellation to work properly? Does it serve any other purpose anyway?

Reply:
Action is a type function defined by this library, I have to use it:
```
// ActionFunc is the action to execute when no subcommands are specified
type ActionFunc func(*Context) error
```
We might want to utilise it in the future for proper cancellation as you mentioned and possibly other stuff, but still even if we don't use it, the most we can do is remove the variable name